### PR TITLE
Issue #2848183 : Implement test for TimestampEventSubscriber.

### PR DESCRIPTION
**Pull request by vasike**

### CODE REVIEW

File: modules/order/tests/src/Kernel/TimestampEventTest.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      'uid' => $user->id(),
+      'ip_address' => '127.0.0.1',
+      'store_id' => $this->store->id(),
+    ]);
+    $order->save();
+
+    $transition = $order->getState()->getTransitions();
+    $order->getState()->applyTransition($transition['place']);
+    $order->save();
+    $this->assertEquals(\Drupal::service('commerce.time')->getRequestTime(), $order->getPlacedTime(), 'During placement transition, the order placed timestamp is set to the request time.');
```
Comment (mglaman) [inline modules/order/tests/src/Kernel/TimestampEventTest.php]: In kernel test, let's use `$this->container->get('commerce.time')`

File: modules/order/src/EventSubscriber/TimestampEventSubscriber.php [hunk 2]
```diff
@@ -27,7 +27,7 @@ public function onPlaceTransition(WorkflowTransitionEvent $event) {
     /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
     $order = $event->getEntity();
     if (empty($order->getPlacedTime())) {
-      $order->setPlacedTime(REQUEST_TIME);
+      $order->setPlacedTime(\Drupal::service('commerce.time')->getRequestTime());
```
Comment (mglaman) [inline modules/order/src/EventSubscriber/TimestampEventSubscriber.php]: Let's inject this via DI

**Review by mglaman — CHANGES_REQUESTED** [review]

File: modules/order/src/EventSubscriber/TimestampEventSubscriber.php (hunk 2 above)
Comment (vasike) [reply modules/order/src/EventSubscriber/TimestampEventSubscriber.php]: done

File: modules/order/tests/src/Kernel/TimestampEventTest.php (hunk 1 above)
Comment (vasike) [reply modules/order/tests/src/Kernel/TimestampEventTest.php]: done. thanks

**Review by mglaman — APPROVED** [review]

**Comment by mglaman** [comment]
Committed in 631ee7d

**Pull request closed without merging**